File: Clases 2018/Clases-C#/ClasePractica11_2018 (update tabla Bugs - asignar bug)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/CriticidadDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class CriticidadDao
    {
        public List<Criticidad> getAll()
        {
            List<Criticidad> lst = new List<Criticidad>();
            foreach (DataRow row in BDHelper.getBDHelper().ConsultaSQL("Select * from Criticidades").Rows)
                lst.Add(map(row));
            return lst;
        }

        private Criticidad map(DataRow row)
        {
            Criticidad oCriticidad = new Criticidad();
            oCriticidad.id_criticidad = Convert.ToInt32(row["id_criticidad"].ToString());
            oCriticidad.nombre = row["n_criticidad"].ToString();
            return oCriticidad;
        }
    }
}

File: PAV 1-TPI - CASOS DE USO version 1/Resueltos/Usuarios+SeleccionActBugs/App_BugTracker resuelto/App_BugTracker/DataLayer/DAOs/CriticidadDao.vb
Public Class CriticidadDao
    Public Function getAll() As List(Of Criticidad)
        Dim lst As New List(Of Criticidad)
        For Each row As DataRow In BDHelper.getDBHelper.ConsultaSQL("Select * from Criticidades").Rows
            lst.Add(map(row))
        Next
        Return lst
    End Function

    Private Function map(ByRef row As DataRow) As Criticidad
        Dim oCriticidad As New Criticidad
        oCriticidad.id_criticidad = Convert.ToInt32(row.Item("id_criticidad").ToString)
        oCriticidad.nombre = row.Item("n_criticidad").ToString
        Return oCriticidad
    End Function
End Class
